vulnerabilities/importers/openssl.py

```diff
@@ -20,101 +20,167 @@
+    ...      <description> OpenSSL 1.0.2 (starting from version 1.0.2b) introduced an "error state"</description>\n\
+    ...      <advisory url="/news/secadv/20171207.txt"/>\n\
+    ...      <reported source="David Benjamin (Google)"/>\n\
+    ... </issue>'
+    >>> issue_parsed = DET.fromstring(issue_string)
+    >>> to_advisory_data(issue_parsed)
+    AdvisoryData(aliases=['VC-OPENSSL-20171207-CVE-2017-3737', 'CVE-2017-3737'], summary='OpenSSL 1.0.2 (starting from version 1.0.2b) introduced an "error state"', affected_packages=[AffectedPackage(package=PackageURL(type='openssl', namespace=None, name='openssl', version=None, qualifiers={}, subpath=None), affected_version_range=OpensslVersionRange(constraints=(VersionConstraint(comparator='=', version=OpensslVersion(string='1.0.2b')), VersionConstraint(comparator='=', version=OpensslVersion(string='1.0.2c')))), fixed_version=OpensslVersion(string='1.0.2n'))], references=[Reference(reference_id='', url='https://github.com/openssl/openssl/898fb884b706aaeb283de4812340bb0bde8476dc', severities=[]), Reference(reference_id='', url='https://www.openssl.org/news/secadv/20171207.txt', severities=[VulnerabilitySeverity(system=ScoringSystem(identifier='generic_textual', name='Generic textual severity rating', url='', notes='Severity for unknown scoring systems. Contains generic textual values like High, Low etc'), value='Moderate')])], date_published=datetime.datetime(2017, 12, 7, 0, 0, tzinfo=datetime.timezone.utc))
+    """
+
+    purl = PackageURL(type="openssl", name="openssl")
+    cve = advisory_url = severity = summary = None
+    safe_pkg_versions = {}
+    vuln_pkg_versions_by_base_version = {}
+    aliases = []
+    references = []
+    affected_packages = []
+    date_published = xml_issue.attrib["public"].strip()
+
+    for info in xml_issue:
+        if info.tag == "impact":
+            severity = VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["generic_textual"], value=info.attrib["severity"]
+            )
+
+        elif info.tag == "advisory":
+            advisory_url = info.attrib["url"]
+            if not advisory_url.startswith("https://web.archive.org"):
+                advisory_url = urljoin("https://www.openssl.org", advisory_url)
+
+        elif info.tag == "cve":
+            cve = info.attrib.get("name")
+            # use made up alias to compensate for case when advisory doesn't have CVE-ID
+            madeup_alias = f"VC-OPENSSL-{date_published}"
+            if cve:
+                cve = f"CVE-{cve}"
+                aliases.append(f"{madeup_alias}-{cve}")
+                aliases.append(cve)
+            else:
+                aliases.append(madeup_alias)
+
+        elif info.tag == "affects":
+            affected_base = info.attrib["base"]
+            affected_version = info.attrib["version"]
+            if affected_base.startswith("fips"):
+                logger.error(f"Unsupported OpenSSL fips. Use a different importer.")
```

Comment:
Will this message be enough to diagnostic and act on when seen in the logs? Try to make it more explicit, add more details and context

Reply:
yeah, need to make that more explicit